### ISSUE TITLE
Latent mode using 2 Lora -> Image quality error

### ISSUE THREAD

**Issue by freebag668** [hako-mikan/sd-webui-regional-prompter#42]
https://postimg.cc/H8Wsc8RS
Images you created previously (Latent mode using 2 Lora)

https://postimg.cc/rd26VT0z
PNG INFO -> Send to txt2img

https://postimg.cc/FfzPXnG1
https://postimg.cc/jwrZndtn
Images created with the same setting
The color of the image is outputting strangely

https://postimg.cc/XZx3Lggn
https://postimg.cc/bSymTQ3f
The output of the image is weird even if you use a simple prompt

https://postimg.cc/m110vv0y
Extension Version
b32d8fe8 (Fri Apr 7 16:05:28 2023)

I format the windows and
I installed a new webui, but it's the same symptom

Is there a solution?

Thank you always for your hard work

**Comment by Symbiomatrix** [comment]
Hello again. So you're saying that latent lora separation broke.
What's the original file's created / modified date?
Do you happen to know which version of webui and the extension you've used to generate it?
I haven't gotten latent mode working properly yet, but if you could try to downgrade the regional extension to the version that worked (according to the file date), perhaps we could pinpoint the faulty commit. It might also be caused by the webui version, but that's more difficult to test, so we'll start with the extension.

**Comment by freebag668** [comment]
> Hello again. So you're saying that latent lora separation broke. What's the original file's created / modified date? Do you happen to know which version of webui and the extension you were using to generate it, before the windows format? I haven't gotten latent mode working properly yet, but if you could try to downgrade the regional extension to the version that worked (according to the file date), we may be able to pinpoint the faulty commit. It might also be caused by the webui version, but that's more difficult to test, so we'll start with the extension.
> 
> Edit: Is it from before or after we fixed #32?

https://github.com/hako-mikan/sd-webui-regional-prompter/issues/32
That's right.
It worked well when it was fixed

**Comment by Symbiomatrix** [comment]
Alright. Can you try this version and see if it works correctly?
https://github.com/Symbiomatrix/sd-webui-regional-prompter/tree/Ddim1

**Comment by freebag668** [comment]
https://github.com/Symbiomatrix/sd-webui-regional-prompter/tree/Ddim1
test
https://postimg.cc/gallery/Y0mgTZ9
https://anonfiles.com/X934Pdj6z9/01_txt
Image quality error


https://github.com/hako-mikan/sd-webui-regional-prompter/tree/b772bfec90670030e95df8ebdf509ade927f6ff5
test
https://postimg.cc/gallery/B3m0TYs
https://anonfiles.com/f461Pajez9/mikan_txt
Image quality error


Image quality errors occur in both versions

**Comment by Symbiomatrix** [comment]
And this one?
https://github.com/hako-mikan/sd-webui-regional-prompter/tree/99ea794c8512caf24f469527d97f06a17d74f6d8

**Comment by freebag668** [comment]
https://github.com/hako-mikan/sd-webui-regional-prompter/tree/99ea794c8512caf24f469527d97f06a17d74f6d8
test
https://postimg.cc/gallery/K2bHFcb
https://anonfiles.com/l8f8W7j7z2/99ea794c8512caf24f469527d97f06a17d74f6d8_txt
Image quality error

**Comment by Symbiomatrix** [comment]
Okay, so more likely caused by a webui version change.
Let's try to compare a simpler prompt, standard dimensions and no hires. I ran this on the [14/3 version](https://github.com/AUTOMATIC1111/stable-diffusion-webui/tree/a9fed7c364061ae6efb37f797b6b522cb3cf7aa2), it probably won't be exactly the same for you as there were some seed breaking changes.
Is the quality of the image below similar to what you've had before?
And does your current version show the same quality degradation for these settings?

![BugRegionQualityB](https://user-images.githubusercontent.com/41131377/230749783-eab55605-d87c-4886-916c-ec969b8a6a05.png)

**Comment by freebag668** [comment]
https://postimg.cc/gallery/2DQGVkh
I set it up and applied it similarly
b32d8fe8 (Fri Apr 7 16:05:28 2023)
Image quality error

**Comment by Symbiomatrix** [comment]
So if even basic settings aren't doing well, I guess it's something to do with the new lora application method. I can't upgrade to that version for now, perhaps you could try downgrading to the previous version,
https://github.com/AUTOMATIC1111/stable-diffusion-webui/tree/a0d07fb5807ad55c8ccfdfc9a6d9ae3c62b9d211
(Unless hako would like to take a shot at it.)

**Comment by freebag668** [comment]
https://github.com/AUTOMATIC1111/stable-diffusion-webui/tree/a0d07fb5807ad55c8ccfdfc9a6d9ae3c62b9d211
test
https://postimg.cc/gallery/jdkJdD2
b9b31c4d (Sun Apr 9 14:38:03 2023)
Image quality error

**Comment by Symbiomatrix** [comment]
Can you please add the debug printout as well?

**Comment by freebag668** [comment]
https://github.com/AUTOMATIC1111/stable-diffusion-webui/tree/a0d07fb5807ad55c8ccfdfc9a6d9ae3c62b9d211
test
https://postimg.cc/gallery/HY2DRdM
https://anonfiles.com/r6N1j2kez1/9d211_txt
b9b31c4d (Sun Apr 9 14:38:03 2023)
Image quality error
debug printout added it

**Comment by Symbiomatrix** [comment]
I think I might finally have a breakthrough. I upgraded to the latest locon extension from my old one (I think it's the 04/03 version, [2e9a79533987e6b454c87705fc4e178febac4fe0](https://github.com/KohakuBlueleaf/a1111-sd-webui-locon/tree/2e9a79533987e6b454c87705fc4e178febac4fe0), and I'm finally getting garbage, with the same settings. It's possible, seeing as some additional lora functions were overridden and that's pretty much the only difference between our debug printouts.
Can you disable / downgrade your locon extension and see if it helps?
![00117-46](https://user-images.githubusercontent.com/41131377/230881735-a90a3093-1856-47e6-a637-24780c974da3.png)

**Comment by hako-mikan** [comment]
@Symbiomatrix 
Thanks for the validation.
In my environment, LoCon extension appeared to be unrelated.
After all, what are the conditions under which this problem occurs?

**Comment by Symbiomatrix** [comment]
@hako-mikan We're attempting to replicate the results in https://postlmg.cc/rd26VT0z . So far, every attempt has fallen short of it - so you might say, any conditions. We know for sure regional's version was https://github.com/hako-mikan/sd-webui-regional-prompter/tree/99ea794c8512caf24f469527d97f06a17d74f6d8 . The webui version is unclear, but we tried either the latest one, or [before the lora change](https://github.com/AUTOMATIC1111/stable-diffusion-webui/tree/a0d07fb5807ad55c8ccfdfc9a6d9ae3c62b9d211), and on my end [14/3](https://github.com/AUTOMATIC1111/stable-diffusion-webui/tree/a9fed7c364061ae6efb37f797b6b522cb3cf7aa2).

Since perfect replication might be impossible (due to various undocumented settings or extension overrides in play), I tried to test a less complex prompt. I'm under the impression that the migurdia lora might be malfunctioning when used in regional, as without easynegative it's showing various forms of discolouration, even with a lower weight, compared to a non regional prompt. Megumin does not suffer the same level of degradation.

Observe:
Regionless, crisp detail.
![BugRegionQualityF](https://user-images.githubusercontent.com/41131377/230964258-e5ed3c97-6b00-4908-8ee1-2b6582c8a557.png)

Migurdia one side (low weight), looks like a picasso whereas other side is unchanged.
![BugRegionQualityE](https://user-images.githubusercontent.com/41131377/230964353-9d30bf56-940b-4e8f-bc39-3247b115dbf5.png)

Migurdia both sides, completely off.
![BugRegionQualityD](https://user-images.githubusercontent.com/41131377/230964540-1264d242-afec-4a3c-ad0b-9974febbcdff.png)

Megumin on standard weight holding up quite well.
![BugRegionQualityG](https://user-images.githubusercontent.com/41131377/230964600-a02f5d65-c902-40a5-b3f4-e14e38b4be8e.png)

**Comment by freebag668** [comment]
disable test
https://postimg.cc/gallery/0byh3Qf
https://anonfiles.com/2dB5w3k3zf/disable_txt
Image quality error


https://github.com/KohakuBlueleaf/a1111-sd-webui-locon/tree/2e9a79533987e6b454c87705fc4e178febac4fe0
downgrade test
https://postimg.cc/gallery/DXFLpcp
https://anonfiles.com/C2D7w4k6z1/c4fe0_downgrade_txt
Image quality error

**Comment by Symbiomatrix** [comment]
@freebag668 Do you have any other loras beside migurdia? Do they show the same quality error?

**Comment by freebag668** [comment]
https://civitai.com/models/23000
https://civitai.com/models/10782
https://civitai.com/models/33990
https://civitai.com/models/11866/meinapastel
This is the model and LORA that I used for the test

Is there anything else I can test?

**Comment by hako-mikan** [comment]
Are the following issues related to this?
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/9118
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/9207

**Comment by Sakura-Luna** [comment]
On earlier versions of the WebUI, the results were also a bit worse.
![蜂蜜浏览器_03264-1435927211-2girls,witch hat,Majestic wizards holding hands and running forward in a magical forest with vibrant colors, detailed hyper-real](https://user-images.githubusercontent.com/53183413/231144624-d406cb1f-c9ec-47aa-93aa-e1f12ce7e5c1.jpg)

**Comment by hako-mikan** [comment]
What happens if you use the settings you're having trouble with and only decreasing strength of LoRA?

**Comment by Sakura-Luna** [comment]
If you just turn off the extension, you can't get such a serious blurred result. If you reduce the weight of both to 0.5, the left side still has a similar bokeh but lighter.

I'm going to try an older version of the extension.

**Comment by Sakura-Luna** [comment]
I reverted to the earliest extended version, no any improvement, you can test Latent Couple + Composable Lora.

Based on my own testing, I think the problem lies with the checkpoints and the Lora model itself, changing any of them can see a significant improvement.

**Comment by freebag668** [comment]
https://postimg.cc/gallery/vs6yZq0
The pictures here are made using a Regional Prompter
You can download the png file to view the information

**Comment by Sakura-Luna** [comment]
After seeing these pictures, I am 90% sure that your results are obtained in Attention mode. I can get similar results.
![蜂蜜浏览器_03265-304826707-2girls,Two women are having a picnic in a beautiful park  They spread out a cozy blanket and set up a delicious spread of sandwi](https://user-images.githubusercontent.com/53183413/231179051-b36c349e-0859-4f0a-863b-25ca7d1aad61.jpg)

**Comment by freebag668** [comment]
https://drive.google.com/file/d/1KzkMXpxL_rsBOAuBHyzqd3B7tYpnm1U_/view?usp=share_link
https://github.com/hako-mikan/sd-webui-regional-prompter/issues/32
This is a video that I filmed while doing a bug test on April 6
If you download the video and watch it closely, the date will be confirmed
Only the first picture has a quality error
It's normal from the second picture

https://postimg.cc/LhS1pmxj
https://postimg.cc/WFpxYHcT
This is a capture when I make a picture
Only the first picture has a photo error
The rest of the picture is normal

Latent mode is used

**Comment by Sakura-Luna** [comment]
What extension is causing so much "up_model" output?

I rolled back the extended version many times, until the Latent just added, and none of the similar results can be reproduced. You can go back to the old version through the extended commit history, and then feedback which version can get the effect you expected, otherwise it is difficult to locate the problem.

**Comment by Symbiomatrix** [comment]
> What extension is causing so much "up_model" output?

That would be regional, on webui 25/3+ version.
```
def changethedevice(module):
    if type(module).__name__ == "LoraUpDownModule":
        if hasattr(module,"up_model") :
            print("up_model")
            module.up_model.weight = torch.nn.Parameter(module.up_model.weight.to(devices.device, dtype = torch.float))
            module.down_model.weight = torch.nn.Parameter(module.down_model.weight.to(devices.device, dtype=torch.float))
        else:
            print("weight")
            module.up.weight = torch.nn.Parameter(module.up.weight.to(devices.device, dtype = torch.float))
            if hasattr(module.down, "weight"):
                module.down.weight = torch.nn.Parameter(module.down.weight.to(devices.device, dtype=torch.float))
```

**Comment by freebag668** [comment]
If I knew more about computers I could help, but I'm sorry I can't.

**Comment by hako-mikan** [comment]
I forgotten deleting `print("up_model")`.
It's for debugging.

**Comment by Symbiomatrix** [comment]
@Sakura-Luna It's an interesting difference, but likely a red herring. I asked freebag to copy his debug printouts earlier in this issue, and he still receives the up_model message along with the degradation.

> https://anonfiles.com/f461Pajez9/mikan_txt

**Comment by hako-mikan** [comment]
@Symbiomatrix
In that case, result will be same, but not same now.
So something wrong.

**Comment by hako-mikan** [comment]
@Symbiomatrix
Not exactly.
It is not exactly the same because I followed the Composable LoRA and erasing LoRA's impact on negative prompts.
Maybe this is where it is affecting.
I will test turning off that feature.

**Comment by hako-mikan** [comment]
No.
When applying LoRA to Textencoder.
Weight of loRA is set to 0.

**Comment by Symbiomatrix** [comment]
Ah, okay. I know that part - te_llist & u_llist's additional segments. So that's what it means.

**Comment by hako-mikan** [comment]
We can disable this feature.
comment out line 1296 
`#regioner.ndeleter()`

Surprisingly, my sample in README get terrible result.
But result of megumin was improved.

Sorry I'm sleepy(2:00 AM now in japan)

**Comment by Symbiomatrix** [comment]
My bad, I hadn't noticed the time. Much obliged. Good night.

**Comment by hako-mikan** [comment]
LoRA settings regarding negative prompts have been added as an option. The default is off; turning it off seems to give better results when using LoRA such as megumin.

**Comment by Sakura-Luna** [comment]
That update was not released to fix your problem.

[Here](https://github.com/hako-mikan/sd-webui-regional-prompter/archive/525741c0532277453c0c7b2359ec7b4a6f710647.zip) is the extension version you used before, you can change the name to `sd-webui-regional-prompter` after decompression, and then copy it to the `extensions` folder under the SD installation location.

**Comment by freebag668** [comment]
> That update was not released to fix your problem.
> 
> [Here](https://github.com/hako-mikan/sd-webui-regional-prompter/archive/525741c0532277453c0c7b2359ec7b4a6f710647.zip) is the extension version you used before, you can change the name to `sd-webui-regional-prompter` after decompression, and then copy it to the `extensions` folder under the SD installation location.

I got it
Thank you always for your hard work

**Comment by Symbiomatrix** [comment]
Hello freebag. Sorry I'm late, but #120  should expand sakura's negative parameters to control individual loras, so you can mess around with those once merged. Wrote an explanation about it in known issues.
